core/dataflow/src/main/java/edu/uci/ics/texera/dataflow/aggregator/Aggregator.java

```diff
@@ -0,0 +1,264 @@
+        switch (aggType) {
+            case COUNT:
+                retVal = true;
+                break;
+
+            case MAX:
+            case MIN:
+                if (Arrays.asList(AttributeType.DATETIME, AttributeType.DATE, AttributeType.STRING, AttributeType.TEXT, AttributeType.INTEGER, AttributeType.DOUBLE).contains(inputAttrType)) {
+                    retVal = true;
+                }
+                break;
+
+            case SUM:
+            case AVERAGE:
+                if (Arrays.asList(AttributeType.INTEGER, AttributeType.DOUBLE).contains(inputAttrType)) {
+                    retVal = true;
+                }
+                break;
+        }
+
+        return retVal;
+    }
+
+    @Override
+    protected void setUp() throws TexeraException {
+        inputSchema = inputOperator.getOutputSchema();
+        List<AggregationAttributeAndResult> aggregationItems = predicate.getAttributeAggregateResultList();
+        for (AggregationAttributeAndResult aggregationItem : aggregationItems) {
+            Schema.checkAttributeExists(inputSchema, aggregationItem.getAttributeName());
+            Schema.checkAttributeNotExists(inputSchema, aggregationItem.getResultAttributeName());
+        }
+
+        for (AggregationAttributeAndResult aggregationItem : aggregationItems) {
+            if (!isAggregationTypeAllowed(inputSchema.getAttribute(aggregationItem.getAttributeName()).getType(), aggregationItem.getAggregatorType())) {
+                throw new TexeraException(
+                        AggregatorErrorMessages.ATTRIBUTE_TYPE_NOT_FIT_FOR_AGGREGATION(aggregationItem.getAttributeName(),
+                                aggregationItem.getAggregatorType().toString()));
+            }
+        }
+
+        outputSchema = transformToOutputSchema(inputSchema);
+    }
+
+    public Schema transformToOutputSchema(Schema... inputSchema) {
+        if (inputSchema.length != 1)
+            throw new TexeraException(String.format(ErrorMessages.NUMBER_OF_ARGUMENTS_DOES_NOT_MATCH, 1, inputSchema.length));
+
+        List<AggregationAttributeAndResult> aggregationItems = predicate.getAttributeAggregateResultList();
+        Builder schemaBuilder = new Schema.Builder();
+        for (AggregationAttributeAndResult aggregationItem : aggregationItems) {
+            schemaBuilder = schemaBuilder.add(aggregationItem.getResultAttributeName(), inputSchema[0].getAttribute(aggregationItem.getAttributeName()).getType());
+        }
+
+        return schemaBuilder.build();
+    }
+
+    @Override
+    protected Tuple computeNextMatchingTuple() throws TexeraException {
+        if (cursor == CLOSED || cursor >= 1) {
+            return null;
+        }
+        Tuple resultTuple = null;
+        List<IField> aggregatedResults = processAllTuples(inputOperator);
+        resultTuple = putResultsIntoTuple(aggregatedResults);
+        cursor++;
+        return resultTuple;
+    }
+
+    private List<IField> processAllTuples(IOperator inputOperator) {
```

Comment:
please add some comments to these private functions, like a overview of what of each these private function do